pilot/utils/exit.py

```diff
@@ -84,6 +101,16 @@ def exit_gpt_pilot(project, ask_feedback=True):
 
     ask_to_store_prompt(project, path_id)
 
-    ask_user_feedback(project, path_id, ask_feedback)
+    # Ask user for contact email. If they choose not to provide it, ask for direct
+    # feedback. Both are optional.
+    if not ask_user_email(project, path_id, ask_feedback):
+        ask_user_feedback(project, path_id, ask_feedback)
+
+    # We can figure out if we're done, but can't distinguish between crash and interrupted, yet
```

Comment:
we can know that in main.py and forward data (eg. exit_info), something like:

except Exception as e:
    exit_info = str(e)
    print(color_red('---------- GPT PILOT EXITING WITH ERROR ----------'))
    traceback.print_exc()
    print(color_red('--------------------------------------------------'))
    ask_feedback = False
except KeyboardInterrupt:
    exit_info = "Execution interrupted by user"
    ask_feedback = False
finally:
    if run_exit_fn:
        exit_gpt_pilot(project, ask_feedback, exit_info)
    sys.exit(0)